File: composeApp/src/androidMain/kotlin/com/xichen/cloudphoto/core/Responsive.kt
package com.xichen.cloudphoto.core

import androidx.compose.runtime.Composable
import androidx.compose.runtime.CompositionLocalProvider
import androidx.compose.runtime.compositionLocalOf
import androidx.compose.ui.platform.LocalConfiguration
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import com.xichen.cloudphoto.core.ui.DeviceInfo
import com.xichen.cloudphoto.core.ui.DeviceType
import com.xichen.cloudphoto.core.ui.ScreenOrientation
import com.xichen.cloudphoto.core.ui.ScreenSize

/**
 * 响应式布局配置
 */
data class ResponsiveConfig(
    val deviceType: DeviceType,
    val screenSize: ScreenSize,
    val orientation: ScreenOrientation,
    val screenWidth: Dp,
    val screenHeight: Dp,
    val isTablet: Boolean,
    val isPhone: Boolean,
    val isLandscape: Boolean,
    val isPortrait: Boolean
)

/**
 * 响应式断点
 */
object Breakpoints {
    val phone: Dp = 600.dp
    val tablet: Dp = 840.dp
    val desktop: Dp = 1200.dp
}

/**
 * Local Responsive Config
 */
val LocalResponsiveConfig = compositionLocalOf<ResponsiveConfig> {
    error("No ResponsiveConfig provided")
}

/**
 * 响应式容器
 */
@Composable
fun ResponsiveContainer(
    content: @Composable () -> Unit
) {
    val configuration = LocalConfiguration.current
    val screenWidth = configuration.screenWidthDp.dp
    val screenHeight = configuration.screenHeightDp.dp
    
    val deviceType = if (screenWidth >= Breakpoints.phone) {
        DeviceType.TABLET
    } else {
        DeviceType.PHONE
    }
    
    val screenSize = when {
        screenWidth < Breakpoints.phone -> ScreenSize.SMALL
        screenWidth < Breakpoints.tablet -> ScreenSize.MEDIUM
        screenWidth < Breakpoints.desktop -> ScreenSize.LARGE
        else -> ScreenSize.EXTRA_LARGE
    }
    
    val orientation = if (screenWidth > screenHeight) {
        ScreenOrientation.LANDSCAPE
    } else {
        ScreenOrientation.PORTRAIT
    }
    
    val config = ResponsiveConfig(
        deviceType = deviceType,
        screenSize = screenSize,
        orientation = orientation,
        screenWidth = screenWidth,
        screenHeight = screenHeight,
        isTablet = deviceType == DeviceType.TABLET,
        isPhone = deviceType == DeviceType.PHONE,
        isLandscape = orientation == ScreenOrientation.LANDSCAPE,
        isPortrait = orientation == ScreenOrientation.PORTRAIT
    )
    
    CompositionLocalProvider(LocalResponsiveConfig provides config) {
        content()
    }
}

/**
 * 获取当前响应式配置
 */
@Composable
fun rememberResponsiveConfig(): ResponsiveConfig {
    return LocalResponsiveConfig.current
}

File: iosApp/iosApp/Core/Responsive.swift
import SwiftUI

/**
 * 响应式配置
 */
struct ResponsiveConfig {
    let deviceType: DeviceType
    let screenSize: ScreenSize
    let orientation: ScreenOrientation
    let screenWidth: CGFloat
    let screenHeight: CGFloat
    
    var isTablet: Bool { deviceType == .tablet }
    var isPhone: Bool { deviceType == .phone }
    var isLandscape: Bool { orientation == .landscape }
    var isPortrait: Bool { orientation == .portrait }
}

enum DeviceType {
    case phone
    case tablet
    case desktop
}

enum ScreenOrientation {
    case portrait
    case landscape
}

enum ScreenSize {
    case small
    case medium
    case large
    case extraLarge
}

/**
 * 响应式环境值
 */
struct ResponsiveConfigKey: EnvironmentKey {
    static let defaultValue: ResponsiveConfig = ResponsiveConfig(
        deviceType: .phone,
        screenSize: .small,
        orientation: .portrait,
        screenWidth: 375,
        screenHeight: 667
    )
}

extension EnvironmentValues {
    var responsiveConfig: ResponsiveConfig {
        get { self[ResponsiveConfigKey.self] }
        set { self[ResponsiveConfigKey.self] = newValue }
    }
}

/**
 * 响应式容器视图
 */
struct ResponsiveContainer<Content: View>: View {
    @Environment(\.horizontalSizeClass) var horizontalSizeClass
    @Environment(\.verticalSizeClass) var verticalSizeClass
    let content: Content
    
    init(@ViewBuilder content: () -> Content) {
        self.content = content()
    }
    
    var body: some View {
        GeometryReader { geometry in
            let config = createConfig(from: geometry)
            content
                .environment(\.responsiveConfig, config)
        }
    }
    
    private func createConfig(from geometry: GeometryProxy) -> ResponsiveConfig {
        let width = geometry.size.width
        let height = geometry.size.height
        
        let deviceType: DeviceType = (horizontalSizeClass == .regular && verticalSizeClass == .regular) ? .tablet : .phone
        let orientation: ScreenOrientation = width > height ? .landscape : .portrait
        
        let screenSize: ScreenSize = {
            let minDimension = min(width, height)
            switch minDimension {
            case ..<600:
                return .small
            case ..<840:
                return .medium
            case ..<1200:
                return .large
            default:
                return .extraLarge
            }
        }()
        
        return ResponsiveConfig(
            deviceType: deviceType,
            screenSize: screenSize,
            orientation: orientation,
            screenWidth: width,
            screenHeight: height
        )
    }
}

/**
 * 响应式配置访问器
 */
extension View {
    func responsiveConfig() -> ResponsiveConfig {
        Environment(\.responsiveConfig).wrappedValue
    }
}

